ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -514,42 +517,46 @@ struct MultiDeviceHostStorage {
         mutable std::mutex shape_mtx;
         MultiDeviceStorage() = default;
 
+        friend void swap(MultiDeviceStorage& first, MultiDeviceStorage& second) {
```

Comment:
`swap` and  move assignment operators should be marked `noexcept`.

Reply:
I added comment. Why I didn't do it in the description.
mutex locks can throw AFAIK.

Reply:
Ah, you're right. I was looking at the constructor of `std::mutex` instead of its `lock` method.